internal/querynode/task.go

```diff
@@ -173,7 +189,11 @@ func (w *watchDmChannelsTask) Execute(ctx context.Context) (err error) {
 	unFlushedSegments := make([]*queryPb.SegmentLoadInfo, 0)
 	unFlushedSegmentIDs := make([]UniqueID, 0)
 	for _, info := range w.req.Infos {
-		for _, ufInfo := range info.UnflushedSegments {
+		unflushedSegmentInfos, err := w.getSegmentInfos(info.UnflushedSegments)
```

Comment:
This requires additional call from QueryNode to DataCoord, which is generally not suggested.

I spotted that `watchDmChannelsTask` is created in QueryCoord, and channelWatchInfo is filled there. So how about simply retrieving segmentIDs while generating `watchDmChannelsTask` in QueryCoord and pass segmentIDs as parameters to QueryNode?

Reply:
Updated.